docs/charmsindetail.md

```diff
@@ -0,0 +1,278 @@
+# Charms in detail
+## Table of contents
+
+[Best practices](##Best%20practice)
+
+[Charm writing in detail](##Charm%20Writing%20in%20detail)
+
+[New to charms?](##New%20to%20charms?)
+
+[Charm Writing](#Charm%20writing)
+
+[Example Charms](##Example%20Charms)
+
+## Best practices
+
+### Repository Naming
+
+Naming follows the convention of charm-<charm_name> for example `charm-percona-cluster`.
+
+The rationale behind this is quick and easy searching of the 1000s of github projects along with other reasons.
+
+## Why charms?
+
+The objective of charms is the following: take the complication of managing 100s potentially 1000s of configuration files and abstract them as python modules. Thus by abstraction creating a true insfrastructure as code.
+
+But charms are not just limited to setting up configuration, they also handle the installation of charms, and can implement actions thus providing not just configuration, but day zero to day two operations support.
+
+One of the ways that charms achieves this goal is through what are called relations, you can think about this much like the Database relation charts you might have seen in previous roles, or at university, the concept is much the same.
+
+Here's an example from the Juju GUI:
+
+![Openstack relations layout](./diagrams/juju_gui.jpg)
+
+What you are looking at here is a Kubernetes, rendered, using Juju and charms. Each of the circles represents an application, each of the lines represents what we call a relation.
+
+More info and history on charms can be [found here](./charmhistory.md).
+
+## Charm writing
+
+To start writing charms you will need to setup a development environment with git and a skeleton charm.
+
+```
+mkdir mynewcharm
+cd $_
+```
+
+Included in the docs directory of this repository is a set of tools, we're going to use the letscharm script to initialise a set of charms for our use.
+
+```
+./tools/letscharm.py
+```
+
+This tool will ask you whether you are creating a K8S charm or a IAAS charm. Pick the correct one as appropriate.
+
+### Add the operator framework
+
+To begin a new charm, you will need to import the operator framework. The charm operator framework is imported by adding the framework in the 'lib' diretory and as a submodule:
+
+```
+git submodule add https://github.com/canonical/operator mod/operator
+ln -s ../mod/operator/ops lib/ops
+```
+
+### Add interface dependences
+
+To begin a new charm we should take note of an Interfaces that might be needed, the operator framework uses Interfaces as dependencies.
```

Comment:
Possible typo here for "any Interfaces"?

Reply:
nice! thanks!